src/types.ts

```diff
@@ -0,0 +1,46 @@
+import { Options } from './index';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// TODO: define what a config is ?
+// interface ResolvedConfig {
+//   [key: string]: unknown;
+// }
+// type ConfigFn = (...params: any[]) => ResolvedConfig;
+// type ConfigFnPromise = (...params: any[]) => Promise<ResolvedConfig>;
+// type ConfigPromise = Promise<ResolvedConfig>;
+
+// https://github.com/davidtheclark/cosmiconfig/issues/134
+export type Config = any;
```

Comment:
Do we want to try and type what we expect the config to be? I don't think we we do but I wanted to check. I think any is best used here.